lua/orgmode/org/mappings.lua

```diff
@@ -295,6 +295,20 @@ function OrgMappings:todo_prev_state()
   self:_todo_change_state('prev')
 end
 
+function OrgMappings:toggle_heading()
+  local node = Files.get_node_at_cursor()
+  local line = vim.fn.getline('.')
+  local parent_level = Files.get_closest_headline().level
+
+  if (node:type() == 'headline' or node:parent():type() == 'headline') then
+    line = line:gsub('^(%*+)', string.rep(' ', parent_level - 1))
+  elseif (node:type() == 'paragraph') then
+    line = line:gsub('^(%s+)', string.rep('*', parent_level + 1) .. ' ')
+  end
```

Comment:
Would it suffice to just check if the `node:type() == 'headline'`?

Some testing in _Emacs_ makes it seem like the only time the headline is toggled _off_ is when you are directly on a headline, otherwise it just adds the leading stars as `closest_headline_star_count + 1`.

Reply:
> what is being done here

My goal is to enable toggling a headline off from anywhere on the actual line. I looked at [this example](https://github.com/milisims/tree-sitter-org#example) and the `node:type()` might not be `headline` but can also be `stars`, `item` or `tags`. Anything that has `headline` as parent. Not evident from the code, but I thought documentation is not really intended in the code in this repo. Correct me if I'm wrong ;)

> when you are directly on a headline

I assume you mean "anywhere on the line of the headline" by that. At least I can do that in my Emacs.

> t just adds the leading stars as `closest_headline_star_count + 1`

I don't know what that means. Could you please elaborate?

Reply:
I think using a simple regex match would be much simpler in this case. There's no need to get treesitter nodes. 
From what I tested in emacs, if current line is a headline, it removes the asterisks, if it's not a headline, it just makes it a child of the parent headline (if any). 
Also, I think there is no need to preserve any indentation. From what I see in emacs, once the asterisks are removed, there is no indentation left, no matter of `org-indent-mode` setting.

Reply:
I went with the way @levouh proposed, checking if the line number of the closest headline is the current line number. Works fine, except for lines that have no closest headline (because they are on the top level). Any idea how to detect those?

@kristijanhusak yeah, unfortunately Emacs doesn't do this. I'll remove the logic then